extensions/amp-autocomplete/0.1/test/validator-amp-autocomplete.out

```diff
@@ -0,0 +1,159 @@
+FAIL
+|  <!--
+|    Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+|   
+|    Licensed under the Apache License, Version 2.0 (the "License");
+|    you may not use this file except in compliance with the License.
+|    You may obtain a copy of the License at
+|   
+|        http://www.apache.org/licenses/LICENSE-2.0
+|
+|    Unless required by applicable law or agreed to in writing, software
+|    distributed under the License is distributed on an "AS-IS" BASIS,
+|    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+|    See the License for the specific language governing permissions and
+|    limitations under the License.
+|  -->
+|  <!--
+|    Test Description:
+|    Tests for the amp-autocomplete tag.
+|  -->
+|
+|  <!doctype html>
+|  <html ⚡>
+|  <head>
+|    <meta charset="utf-8">
+|    <title>amp-autocomplete example</title>
+|    <link rel="canonical" href="amps.html">
+|    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+|    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+|    <script async custom-element="amp-autocomplete" src="https://cdn.ampproject.org/v0/amp-autocomplete-0.1.js"></script>
+|    <script async custom-element="amp-form" src="https://cdn.ampproject.org/v0/amp-form-0.1.js"></script>
+|    <script async custom-element="amp-bind" src="https://cdn.ampproject.org/v0/amp-bind-0.1.js"></script>
+|    <script async src="https://cdn.ampproject.org/v0.js"></script>
+|  </head>
+|  <body>
+|    <!-- Valid: amp-component with static data and filter=prefix attribute -->
+|    <amp-autocomplete filter="prefix">
+>>   ^~~~~~~~~
+amp-autocomplete/0.1/test/validator-amp-autocomplete.html:36:2 The implied layout 'CONTAINER' is not supported by tag 'amp-autocomplete'. (see https://www.ampproject.org/docs/reference/components/amp-autocomplete) [AMP_LAYOUT_PROBLEM]
+|      <input type="text">
+|      <script type="application/ld+json">
```

Comment:
This should be `type="applciation/json"`. `ld+json` is [for a different purpose](https://en.wikipedia.org/wiki/JSON-LD).

Reply:
The validator is outputting a new message "Custom JavaScript is not allowed" when I make this change. Is this something I can update in the validator rules to make it allowed?

Reply:
I added a child_tag spec for the `input` and `script` tags as children to `amp-autocomplete`. With this I was able to get rid of the "Custom JavaScript is not allowed" invalidations. 

Now there is one case remaining that is not showing as invalid that should be: when `<script>` is included but `<input>` is not. `<input>` should always be a child of autocomplete and `<script>` can sometimes be, but I'm not sure how to specify one as always mandatory and the other as optional. From the convention I am using it makes it mandatory to have at least one of the two but it doesn't matter which.

Reply:
Hi! I work on validation for AMP and it'd be helpful to have some more background on this extension and what it's trying to do so we can create precise validation rules. I noticed that `amp-autocomplete` requires the extension `amp-form`, but it doesn't have to be a child of `amp-form`. Is `amp-autocomplete` something that still operates without a `form` element? The markdown file doesn't show `form`  either. Is there an i2i with more details than the markdown contains?

Reply:
The idea is that amp-autocomplete should always be a child of `form` because it should always be a parent of `input`, so my apologies for not specifying that clearly in the rules. 

You can find the I2I at #20988. Thanks for your help! 😄